subcommand/mesh-init/command.go

```diff
@@ -63,31 +60,68 @@ func (c *Command) Run(args []string) int {
 }
 
 func (c *Command) realRun() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	taskMeta, err := awsutil.ECSTaskMetadata()
 	if err != nil {
 		return err
 	}
 
 	cfg := api.DefaultConfig()
 
+	consulClient, err := api.NewClient(cfg)
+	if err != nil {
+		return fmt.Errorf("constructing consul client: %s", err)
+	}
+
+	serverConnMgrCfg, err := c.config.ConsulServerConnMgrConfig(taskMeta)
+	if err != nil {
+		return fmt.Errorf("constructing server connection manager config: %s", err)
+	}
+	serverConnMgrCfg.ServerWatchDisabled = false
+
+	watcher, err := discovery.NewWatcher(ctx, serverConnMgrCfg, c.log)
+	if err != nil {
+		return fmt.Errorf("unable to create consul server watcher: %s", err)
+	}
+
+	go watcher.Run()
+	defer watcher.Stop()
+
+	state, err := watcher.State()
+	if err != nil {
+		return fmt.Errorf("unable to fetch consul server watcher state: %s", err)
+	}
+
 	if c.config.ConsulLogin.Enabled {
-		// If enabled, login to the auth method to obtain an ACL token.
+		// If enabled write the ACL token to a shared volume so that consul-dataplane
+		// can reuse it later on whenever it starts up
 		tokenFile := filepath.Join(c.config.BootstrapDir, config.ServiceTokenFilename)
-		if err := c.loginToAuthMethod(tokenFile, taskMeta); err != nil {
+		err = os.WriteFile(tokenFile, []byte(state.Token), 0644)
+		if err != nil {
 			return err
 		}
+
+		// Temporary workaround so that unit tests relying on the
+		// previous version of the client (that talks through agents)
+		// can pass.
 		cfg.TokenFile = tokenFile
 
-		// The just-created token is not immediately replicated to Consul server followers.
-		// Mitigate against this by waiting for the token in stale consistency mode.
-		if err := c.waitForTokenReplication(tokenFile); err != nil {
-			return err
-		}
+		c.log.Info("successfully logged in", "token-file", tokenFile)
```

Comment:
minor: The server-connection-manager will [log about the ACL login](https://github.com/hashicorp/consul-server-connection-manager/blob/main/discovery/watcher.go#L445), so this log line could be changed to be more specific:

```suggestion
		c.log.Info("wrote ACL token to shared volume", "token-file", tokenFile)
```